charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -169,6 +169,7 @@ receivers:
     collection_interval: 20s
     auth_type: "serviceAccount"
     endpoint: "${env:K8S_NODE_NAME}:10250"
+    insecure_skip_verify: true
```

Comment:
I don't think we want to enable this by default since it isn't the recommended behavior for production.  Really the receiver's README should be updated to not set that value. If the cluster's certificates are in order this value is not necessary.  @jinja2 do you agree?

Reply:
Also @povilasv and @dmitryax

Reply:
At the moment the `preset` does not add any of the `ca_file`, `key_file`, `cert_file`, right? So I wonder in which cases the preset is actually useful with the current implementation.

ref: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27070/files#r1334828771

Reply:
Someone correct me if I'm wrong, but I thought the receiver was looking in [an expected place for the certs](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e1403742f39d090bfc3c164f5a295d1783df3fb1/internal/kubelet/client.go#L25-L26) and `ca_file`, `key_file`, and `cert_file` was only necessary if your certs didn't live in that expected place.

Are the certs for a standard production k8s cluster no longer installed at that location?

Reply:
The preset in the chart is using the `serviceAccount` auth type. This auth type does not use the `ca_file`, `key_file`, `cert_file` options at all. These are used for auth type = TLS so basically if you are using client certificates to authenticate. With serviceAccount auth we get the cluster's root CA cert so it can be added to the trust store/cert pool, and this is the `ca.crt` we see along with the token for service account at the well-known path that Tyler linked. The standard practice is to setup kubelet to serve certificate signed by the cluster CA which we have access to already when using serviceaccount auth.

Some setups might not set the k8s node name in the Subject alternative name in the kubelet's server cert. In such cases users might get the `hostname mismatch` error since we use the K8S_NODE_NAME as host here. The referenced [comment]( https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27070/files#r1334828771) has more details on my findings on some of the more commonly used distros.

I would recommend to not add `insecure_skip_verify: true` as default cause that would be a step back in terms of secure practices. This is the first issue about this setting after the change in behavior went out so it might not be impacting most users. A compromise might be an option under the preset variable?

The error posted in the issue with running in kind cluster can most likely be fixed by adding below patch to the kind cluster's config.
```
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
kubeadmConfigPatches:
  - |
    kind: KubeletConfiguration
    serverTLSBootstrap: true
 ```
 After the cluster is up, you'll probably need to approve the certificate signing request for the kubelet certs by running `kubectl certificate approve {PENDING CSR}`.

Reply:
Thank's for the details @jinja2 @TylerHelmuth on how the internals of `serviceAccount` auth type work.

Based on what you shared `insecure_skip_verify: true` does not make a lot of sense to be the default.
I tested the preset on a GKE cluster and works without any issues.
However would that make sense to document these details around the `serviceAccount` type so as to make it clear how it works and what is required in order for this work?

Reply:
@ChrsMark Yes, let's definitely document it and the possible fix.